modules/ROOT/pages/dataweave-pattern-matching.adoc

```diff
@@ -1,39 +1,109 @@
-= Pattern Matching in DataWeave
+= Pattern Matching in DataWeave Through match Statements
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 
-In DataWeave, pattern matching executes on the first pattern that matches the specified expression. DataWeave supports four different types of patterns:
+The `match` function behaves like a `match` or `switch` statement in other
+languages, like Java or C++, and routes an input expression to a particular
+output expression based on some conditions.
 
+.Match Statement Syntax:
+[source,dataweave,linenums]
+----
+inputExpression match {
+  case <condition> -> <routing expression>
+  case <condition> -> <routing expression>
+  else -> <default routing expression>
+}
+----
+
+As in other languages, the DataWeave `match` function provides compact way to
+organize multiple, chained `if-else` statements. A `match` expression consists
+of a list of `case` statements that optionally end with an `else` statement.
+Each `case` statement consists of a conditional selector expression that must
+evaluate to `true` or `false`. This conditional expression corresponds to the
+condition expressed by an `if` statement, followed by a corresponding DataWeave
+routing expression that can include the `match` function's input expression.
+
+Each `case` statement can be an `if` statement, or the `case` statement can use
+other pattern-matching shortcuts to define the `case` statement's condition. If
+a `case` statement is `false`, its routing expression is ignored.
+
+DataWeave supports four different types of patterns for a `case` statement's
+condition:
+
+* <<pattern_match_expressions,Expressions>>
 * <<pattern_match_literals,Literal values>>
 * <<pattern_match_data_types,Data types>>
-* <<pattern_match_expressions,Expressions>>
 * <<pattern_match_regex,Regular expressions>>
 
-Each pattern type can be named or unnamed.
+Each case can be named or unnamed. If the case is named, the name stores the
+input expression as a local variable that can be used both in that case
+statement's conditional expression and in the corresponding routing expression.
 
-.Pattern Matching Model
+.Match Statement Structure
```

Comment:
In an earlier title, you had "match Statement".

Reply:
fixing